File: app/src/main/java/com/example/learnwordstrainer/integration/OpenAIAPI.java
package com.example.learnwordstrainer.integration;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface OpenAIAPI {
    @POST("v1/chat/completions")
    Call<ChatCompletionResponse> getChatCompletion(
            @Header("Authorization") String authHeader,
            @Body ChatCompletionRequest body
    );
}

File: app/src/main/java/com/shastkiv/vocab/data/remote/api/OpenAIAPI.kt
package com.shastkiv.vocab.data.remote.api

import com.shastkiv.vocab.data.remote.dto.ChatCompletionRequest
import com.shastkiv.vocab.data.remote.dto.ChatCompletionResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.Header
import retrofit2.http.POST

interface OpenAIAPI {
    @POST("v1/chat/completions")
    suspend fun getChatCompletion(
        @Header("Authorization") authHeader: String,
        @Body body: ChatCompletionRequest
    ): Response<ChatCompletionResponse>
}